ansible/roles/block_devices/library/block_devices.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2021, StackHPC
+# Apache 2 License
+
+DOCUMENTATION = r'''
+---
+module: block_devices
+
+short_description: Return block device paths by serial number.
+
+options: (none)
+
+author:
+    - Steve Brasier (@sjpb)
+'''
+
+RETURN = r'''
+devices:
+    description: dict with device serial numbers as keys and full paths (e.g. /dev/sdb) as values
+    type: dict
+    return: always
+'''
+
+import json
+
+from ansible.module_utils.basic import AnsibleModule
+
+def run_module():
+    module_args = dict()
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)
+    result = {"changed": False}
+    _, stdout, _ = module.run_command("lsblk --paths --json -O", check_rc=True)
```

Comment:
NB `json` only works on CentOS 8 but that's ok

Reply:
And so does `-O`

Reply:
Something like `lsblk --paths --list -o name,serial` works on CentOS7 (although obvs format is different)